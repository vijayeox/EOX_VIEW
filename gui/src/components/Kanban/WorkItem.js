import React from "react";
import "./WorkGroup.scss";
import { Container, ListGroup } from "react-bootstrap";
import { Draggable } from "react-beautiful-dnd";
import { Card, CardImg, CardBody } from "reactstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
//import { faHistory } from "@fortawesome/fontawesome-free-solid";

const dateConvert = date => {
  const myDate = new Date(date * 1000);

  return myDate.toLocaleDateString("en-Us", {
    month: "long",
    day: "2-digit",
    year: "numeric",
  });
};

export default function WorkItem(props) {
  const CardInfo = props.cardInfo;

  return (
      <Draggable
        key={CardInfo.myId}
        draggableId={CardInfo.uuid}
        index={props.index}
      >
        {(provided, snapshot) => {
          return (
            <ListGroup.Item
              variant="dark"
              ref={provided.innerRef}
              {...provided.draggableProps}
              {...provided.dragHandleProps}
              style={{
                userSelect: "none",
                minHeight: "50px",
                padding: "0",
                border: "0",
                backgroundColor: snapshot.isDragging ? "white" : "white",
                ...provided.draggableProps.style,
              }}
            >
              <Card className="taskcards k_taskcards" style={{ border: "0" }}>
                <CardBody
                  style={{
                    width: "100%",
                    top: "1rem",
                    padding: "0",
                    border: "0",
                  }}
                >
                  <div className="k_taskcardDisplay">
                    <CardImg
                      style={{
                        margin: "5px",
                        padding: "auto",
                        borderRadius: "50%",
                      }}
                      className="img k_img"
                      src="https://source.unsplash.com/user/c_v_r/100x100"
                    />
                    {/*<FontAwesomeIcon
                    size="1.5x"
                    icon={['fal', 'user-circle']}
                     /> */}

                    <p className="k_taskcard-p">{CardInfo.created_by}</p>
                  </div>
                  <div>
                    <label className="k_taskcard-label">
                      <label className="k_taskcard-sublabel"> Due:</label>{" "}
                      {CardInfo.date_created.substring(0, 10)}
                    </label>
                    <FontAwesomeIcon
                      style={{
                        color: CardInfo.rygStatus,
                        float: "right",
                        fontWeight: 'bold'
                      }}
                      size="1x"
                      icon={['fal', 'history']}
                    />
                  </div>
                </CardBody>
              </Card>
            </ListGroup.Item>
          );
        }}
      </Draggable>
  );
}
