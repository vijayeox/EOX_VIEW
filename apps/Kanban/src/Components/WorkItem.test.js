import React from 'react';
import { shallow, configure,mount } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import WorkItem from './WorkItem';
configure({ adapter: new Adapter() })


describe('Work Item ',()=>{
    const itemData = 
    [
        {
            itemId: 'item-100',
            workGroupId: "uuid1",
            itemName: 'Person-1',
            itemIcon: 'abc',
            itemDescription: 'General Manager',
            timestamp: 1622575446,
            priority: 'High',
            img:'./profile1.jpeg'
        },
    ]
    it('render Correctly',()=>{
        //const store = mockStore(itemData);
        const wrapper=shallow(<WorkItem CardInfo={itemData}/>)
        expect(wrapper).toMatchSnapshot()
    })
    it('Component is not empty',()=>{
        const wrapper=shallow(<work />)
        expect(wrapper.exists()).toBe(true)
    });
    it('props value',()=>{
        const wrapper=shallow(<work   CardInfo={itemData}/>)

        expect(wrapper).toHaveLength(itemData.length)
    })

})

// beforeEach(() => wrapper = shallow(
//     <WorkItem
    
//     />
// ));

// test('render correctly',()=>{
//    // const wrapper=shallow(<WorkItem/>)
//     expect(wrapper.exists()).toBe(true)

// }