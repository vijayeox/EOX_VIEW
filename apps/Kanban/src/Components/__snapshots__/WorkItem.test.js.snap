// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Work Item  render Correctly 1`] = `ShallowWrapper {}`;
